api/Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM alpine:3.12.3
+
+RUN apk add --update --no-cache \
+  build-base \
+  nodejs-current \
+  curl \
+  npm \
+  tzdata \
+  libxslt \
+  bash \
+  jq    \
+  git
+
+WORKDIR /app
+
+COPY ./api ./api
+
+RUN cd api && npm ci --verbose
+COPY ./shared-libs ./api/node_modules/@medic
+COPY ./api/src/public/ /app/api/build/public/
+
+SHELL ["/bin/bash", "-c"]
+RUN for i in $(ls -d /app/api/node_modules/@medic/*); \
+                do \
+                cd "$i" && \
+                npm ci && \
+                echo "$i"; done
+
+ENV NODE_PATH=/app/api/node_modules
+
+RUN npm install -g horticulturalist
```

Comment:
Can this move to before all the api work (essentially before line 18)? If so, it would allow docker to cache/skip this step most of the time, where at it's current location, it'll always rerun since it follows the api parts.

...I may be getting my dockerfile cache logic mixed up though, so this comment is mostly a check.

Reply:
I'm not sure if we'll need horti, depending on what devs want to do with it.

You are certainly right about docker layer caching :) There's a lot we want to optimize, but admittedly makes debugging more tedious. Copying just package.json and building node_modules will let that cache as a layer. Using multi-layer approach and building a base and pulling that base layer into sentinel lets us keep one common docker cached layer of apk packages. I'll bring these in.